__tests__/utils.test.ts

```diff
@@ -369,6 +369,14 @@ describe('Test secret parsing and handling', () => {
         expect(transformToValidEnvName('0Admin')).toBe('_0ADMIN')
     });
 
+    test('Maintains single underscore between prefix and numeric properties', () => {
```

Comment:
Testing this functionality at a higher lever via `injectSecret()` would be better if we can.